db/migrate/20210512192745_fix_red_hat_root_repository_arch.rb

```diff
@@ -1,8 +1,18 @@
 class FixRedHatRootRepositoryArch < ActiveRecord::Migration[6.0]
   def up
-    ::Katello::RootRepository.
-      joins("INNER JOIN katello_contents ON katello_contents.cp_content_id = katello_root_repositories.content_id").
-      where.not(arch: 'noarch').where.not("katello_contents.content_url ILIKE '%$basearch%'").update(arch: 'noarch')
+    ::Katello::RootRepository.class_eval do
+      def upstream_authentication_token=(_)
+        nil
+      end
+
+      def upstream_authentication_token_changed?
+        false
+      end
```

Comment:
one common way that we handle this is by defining a fake class, like:

https://github.com/Katello/katello/blob/ea9570ae2e2529b54f20f113abdab86b83371040/db/migrate/20180920214134_create_repository_root.rb#L46-L48

this way the class uses the same table, but has no validations/hooks

Reply:
the benefit is that it doesn't modify the class which could affect other migrations down the line